massa-network/src/peer_info_database.rs

```diff
@@ -876,24 +814,15 @@ impl PeerInfoDatabase {
     /// Sorts peers by ( last_failure, rev(last_success) )
     /// and returns as many peers as there are available slots to attempt outgoing connections to.
     pub fn get_out_connection_candidate_ips(&self) -> Result<Vec<IpAddr>, NetworkError> {
-        Ok([
-            self.get_out_connection_candidate_ips_for_type(
-                PeerType::Bootstrap,
-                &self.bootstrap_connection_count,
-                &self.network_settings.bootstrap_peers_config,
-            )?,
-            self.get_out_connection_candidate_ips_for_type(
-                PeerType::WhiteListed,
-                &self.whitelist_connection_count,
-                &self.network_settings.whitelist_peers_config,
-            )?,
-            self.get_out_connection_candidate_ips_for_type(
-                PeerType::Standard,
-                &self.standard_connection_count,
-                &self.network_settings.standard_peers_config,
-            )?,
-        ]
-        .concat())
+        let mut connections = vec![];
+        for (peer_type, connection_count) in self.peer_types_connection_count.iter() {
```

Comment:
Here the order matter. Do you know if it's guaranteed ?

Reply:
It was linked to the order of creation but better sort to be sure just added it.